docs/codeql/writing-codeql-queries/about-codeql-queries.rst

```diff
@@ -114,6 +115,8 @@ You can modify the alert message defined in the final column of the ``select`` s
 
 Select clauses for path queries (``@kind path-problem``) are crafted to display both an alert and the source and sink of an associated path graph. For more information, see ":doc:`Creating path queries <creating-path-queries>`."
 
+Select clauses for diagnostic queries (``@kind diagnostic``) have different requirements. For examples, see the `diagnostic queries in the CodeQL repository <https://github.com/github/codeql/search?q=%22%40kind+diagnostic%22>`__.
```

Comment:
You may want to add a similar link for summary metric queries, though the search is a little harder to phrase because there are queries with `@kind metric` but no `@tags summary`, and we don't want to include those. https://github.com/github/codeql/blob/main/cpp/ql/src/Summary/LinesOfCode.ql is an example.